### DISCUSSION TITLE
Changing Temp Directory

### DISCUSSION THREAD

**Discussion by hardcod3dd** [pawelsalawa/sqlitestudio#3705] [Q&A]
### Details
database or disk is full error happens.

### Steps to reproduce
create index

### Operating system
windows 10 latest build.x64

### SQLiteStudio version
3.2.1

I have 2 disk.1 is ssd 60gb empty space and 1 600gb empty space ssd.whenever i try to create index.it writes to disk c: so i run out of disk space and process fails.is there any way to change directory to my disk d:?
database is 174gb in size.

**Comment by hardcod3dd** [comment]
Solved by changing windows temp path.but is there any option on software ? without changing whole operationg systems temp path?

**Comment by pawelsalawa** [comment]
Are you placing your database file on that bigger disk?

**Comment by hardcod3dd** [comment]
the database is in big drive.yes. but temporary files still written to main windows disk.

also i started indexing about 19 hours ago and still indexing is it normal ? it created 1 100gb and 1 4gb file in temp directory
database is 174 gb

disk usage is about 4-5Mb/s

**Comment by pawelsalawa** [comment]
This isn't really much of SQLiteStudio's thing. This is how SQLite itself works.
This pragma can control it a little bit: https://sqlite.org/pragma.html#pragma_temp_store
You get to choose if you want to use file to store it, or a memory.

Below that pragma there is another one, which controls the directory where temp files are stored, although this is pretty much what you do by changing system's temp path and my best guess is that this is a correct way to control the temp path in SQLite.

You can always ask at https://dba.stackexchange.com/

**Comment by pawelsalawa** [comment]
As for the time of indexing - I'm assuming you just create an index on some huge table.
I guess it can take that time. I'm not sure. To be honest I have never done it on such big table. This - again - is more of SQLite concern, rather than SQLiteStudio. I think it's a good question for https://dba.stackexchange.com/
